corehq/apps/api/resources/v0_5.py

```diff
@@ -121,10 +119,7 @@ def _set_role_for_bundle(kwargs, bundle):
         qualified_role_id = domain_roles[0].get_qualified_id()  # roles may not be unique by name
         bundle.obj.set_role(kwargs['domain'], qualified_role_id)
     else:
-        # check for preset roles and now create them for the domain
-        preset_role_id = UserRolePresets.get_preset_role_id(bundle.data.get('role'))
-        if preset_role_id:
-            bundle.obj.set_role(kwargs['domain'], preset_role_id)
+        raise BadRequest(f"Invalid User Role '{bundle.data.get('role')}'")
```

Comment:
We won't allow preset roles in api now?
`by_domain_and_name` would return only roles present in the DB.

Reply:
Roles are created in the DB when the domain is created.